src/gfx/pal_spec.cpp

```diff
@@ -88,25 +89,31 @@ void parseInlinePalSpec(char const * const rawArg) {
 	for (;;) {
 		++n; // Ignore the '#' (checked either by caller or previous loop iteration)
 
-		Rgba &color = options.palSpec.back()[nbColors];
-		auto pos = std::min(arg.find_first_not_of("0123456789ABCDEFabcdef"sv, n), arg.length());
-		switch (pos - n) {
-		case 3:
-			color = Rgba(singleToHex(arg[n + 0]), singleToHex(arg[n + 1]), singleToHex(arg[n + 2]),
-			             0xFF);
-			break;
-		case 6:
-			color = Rgba(toHex(arg[n + 0], arg[n + 1]), toHex(arg[n + 2], arg[n + 3]),
-			             toHex(arg[n + 4], arg[n + 5]), 0xFF);
-			break;
-		case 0:
-			parseError(n - 1, 1, "Missing color after '#'");
-			return;
-		default:
-			parseError(n, pos - n, "Unknown color specification");
-			return;
+		std::optional<Rgba> &color = options.palSpec.back()[nbColors];
+		// Check for #none first.
+		if (arg.compare(n, 4, "none"sv) == 0 || arg.compare(n, 4, "NONE"sv) == 0) {
```

Comment:
Looks like this would allow you to omit the comma after `none`: e.g. `-c "#FFF,#none#none#000"`.

Reply:
Isn't the comma checked afterwards? The `else` branch doesn't check for it either.

Reply:
Yup, this code is correct
```
$ ./rgbgfx ~/nonetest.png -c#000,#none#none,#fff
error: Unexpected character, expected ',', ';', or end of argument
In inline palette spec: #000,#none#none,#fff
                                  ^
Conversion aborted after 1 error
```